cognite/neat/_rules/analysis/_base.py

```diff
@@ -65,271 +61,258 @@ def to_pandas(self) -> pd.DataFrame:
-                different namespaces or not. Defaults False
+            include_ancestors (bool, optional): Include ancestors of the parents. Defaults to False.
+            include_different_space (bool, optional): Include parents from different spaces. Defaults to False.
 
         Returns:
-            Dictionary of classes with a list of properties defined for them
-
-        !!! note "consider_inheritance"
-            If consider_inheritance is True, properties from parent classes will also be considered.
-            This means if a class has a parent class, and the parent class has properties defined for it,
-            while we do not have any properties defined for the child class, we will still consider the
-            properties from the parent class. If consider_inheritance is False, we will only consider
-            properties defined for the child class, thus if no properties are defined for the child class,
-            it will not be included in the returned dictionary.
+            dict[ClassEntity, set[ClassEntity]]: Values parents with class as key.
         """
+        parents_by_class: dict[ClassEntity, set[ClassEntity]] = {}
+        for class_ in self.information.classes:
+            parents_by_class[class_.class_] = set()
+            for parent in class_.implements or []:
+                if include_different_space or parent.prefix == class_.class_.prefix:
+                    parents_by_class[class_.class_].add(parent)
+                else:
+                    warnings.warn(
+                        NeatValueWarning(
+                            f"Parent class {parent} of class {class_} is not in the same namespace, skipping!"
+                        ),
+                        stacklevel=2,
+                    )
+        if include_ancestors:
+            self._include_ancestors(parents_by_class)
 
-        class_property_pairs: dict[T_ClassEntity, list[T_Property]] = defaultdict(list)
-
-        for property_ in self._get_properties():
-            class_property_pairs[self._get_cls_entity(property_)].append(property_)  # type: ignore
-
-        if consider_inheritance:
-            class_parent_pairs = self.class_parent_pairs(allow_different_namespace)
-            for class_ in class_parent_pairs:
-                self._add_inherited_properties(class_, class_property_pairs, class_parent_pairs)
+        return parents_by_class
 
-        return class_property_pairs
+    @staticmethod
+    def _include_ancestors(parents_by_class: dict[T_Hashable, set[T_Hashable]]) -> None:
+        # Topological sort to ensure that classes include all ancestors
+        for class_entity in list(TopologicalSorter(parents_by_class).static_order()):
+            parents_by_class[class_entity] |= {
+                grand_parent for parent in parents_by_class[class_entity] for grand_parent in parents_by_class[parent]
+            }
```

Comment:
Not the last method did not do this. That could lead to only including the immediate parent. This guarantees that all parents are included.